storm-core/src/jvm/org/apache/storm/utils/Utils.java

```diff
@@ -1382,5 +1422,438 @@ public static TopologyInfo getTopologyInfo(String name, String asUser, Map storm
+                targetFilename);
+    }
+
+    /**
+     * Returns a Collection of file names found under the given directory.
+     * @param dir a directory
+     * @return the Collection of file names
+     */
+    public static Collection<String> readDirContents(String dir) {
+        Collection<String> ret = new HashSet<>();
+        File[] files = new File(dir).listFiles();
+        if (files != null) {
+            for (File f: files) {
+                ret.add(f.getName());
+            }
+        }
+        return ret;
+    }
+
+    /**
+     * Returns the value of java.class.path System property. Kept separate for
+     * testing.
+     * @return the classpath
+     */
+    public static String currentClasspath() {
+        return _instance.currentClasspathImpl();
+    }
+
+    public String currentClasspathImpl() {
+        return System.getProperty("java.class.path");
+    }
+
+    /**
+     * Returns a collection of jar file names found under the given directory.
+     * @param dir the directory to search
+     * @return the jar file names
+     */
+    private static Collection<String> getFullJars(String dir) {
+        File[] files = new File(dir).listFiles(new FilenameFilter() {
+            @Override
+            public boolean accept(File dir, String name) {
+                return name.endsWith(".jar");
+            }
+        });
+        Collection<String> ret = new HashSet<String>(files.length);
+        for (File f : files) {
+            ret.add(Paths.get(dir, f.getName()).toString());
+        }
+        return ret;
+    }
+
+    public static String workerClasspath() {
+        String stormDir = System.getProperty("storm.home");
+        String stormLibDir = Paths.get(stormDir, "lib").toString();
+        String stormConfDir =
+                System.getenv("STORM_CONF_DIR") != null ?
+                System.getenv("STORM_CONF_DIR") :
+                Paths.get(stormDir, "conf").toString();
+        String stormExtlibDir = Paths.get(stormDir, "extlib").toString();
+        String extcp = System.getenv("STORM_EXT_CLASSPATH");
+        if (stormDir == null) {
+            return Utils.currentClasspath();
+        }
+        List<String> pathElements = new LinkedList<>();
+        pathElements.addAll(Utils.getFullJars(stormLibDir));
+        pathElements.addAll(Utils.getFullJars(stormExtlibDir));
+        pathElements.add(extcp);
+        pathElements.add(stormConfDir);
+
+        return StringUtils.join(pathElements,
+                System.getProperty("path.separator"));
+    }
+
+    public static String addToClasspath(String classpath,
+                Collection<String> paths) {
+        return _instance.addToClasspathImpl(classpath, paths);
+    }
+
+    public String addToClasspathImpl(String classpath,
+                Collection<String> paths) {
+        if (paths == null || paths.isEmpty()) {
+            return classpath;
+        }
+        List<String> l = new LinkedList<>();
+        l.add(classpath);
```

Comment:
In the previous code it would return "nil:a:b:c"  now it will return "null:a:b"  In this case I am not sure it makes a difference.